src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -129,19 +135,28 @@ public void insertRecordToBuffer(SinkRecord record) {
     }
   }
 
-  /* TODO: SNOW-529753 Deal with rebalances and restarts */
-  /* TODO: SNOW-536429 For EOS, also fetch the offset from snowflake during rebalance/restart */
-  private void init(long recordOffset) {
+  /* For EOS, fetch the offset from snowflake during only during first time after connector starts and record is received in Kafka Connector */
+  private void fetchLastCommittedOffsetToken() {
+    String offsetToken = this.channel.getLatestCommittedOffsetToken();
     try {
-      startCleaner(recordOffset);
-    } catch (Exception e) {
-      LOGGER.error("Cleaner and Flusher threads shut down before initialization");
+      if (offsetToken == null) {
+        this.offsetPersistedInSnowflake.set(-1);
+      } else {
+        this.offsetPersistedInSnowflake.set(Long.parseLong(offsetToken));
+      }
+      LOGGER.info(
+          "Fetched offsetToken:{} for channelName:{} during channel initialization.",
+          this.offsetPersistedInSnowflake.get(),
+          this.getChannelName());
+    } catch (NumberFormatException e) {
```

Comment:
Should we throw an exception here? Silently continuing is a bit scary IMO

Reply:
This shouldn't happen, and is just a cautious check. But yes, we should fail starting the connector if this happens.